### PR TITLE
feat: repairnator jenkins plugin uses subprocesses

### PR DESCRIPTION
**Pull request by henry-lp**
Subprocess version of the plugin (#469)

### CODE REVIEW

File: repairnator/pom.xml [hunk 1]
```diff
@@ -61,7 +61,6 @@
         <module>repairnator-pipeline</module>
         <module>repairnator-core</module>
         <module>repairnator-realtime</module>
-        <module>repairnator-jenkins-plugin</module>
```
Comment (monperrus) [inline repairnator/pom.xml]: why deleting?

Comment (henry-lp) [reply repairnator/pom.xml]: It makes the delivery phase after merging with master fails due to `401 code` per discussion (root try to stage in `Sonatype` and the does not allow `.hpi` I think) . I seems to manage to deploy this as a separate project , so maybe I should try to `cd` into `repairnator-jenkins-plugin` and `mvn deploy` after deploying the usual repairnator from root ?

Comment (monperrus) [reply repairnator/pom.xml]: OK, we can keep it this way, but then we need to modify the travis config to also build this project, as we do for maven-repair

Comment (henry-lp) [reply repairnator/pom.xml]: Yes , It is done :) .

**Comment by henry-lp** [comment]
ready fo reviews :) .

File: .travis/travis-deploy.sh [hunk 2]
```diff
@@ -14,6 +14,10 @@ if [ "$TRAVIS_PULL_REQUEST" = "false" ] && [ "$TRAVIS_BRANCH" = "master" ]; then
     sed -i -e 's/\${revision}/'$VERSION'/' pom.xml */pom.xml
     git diff
     mvn -q deploy -DskipTests -Dcheckstyle.skip
+
+    # push jenkins plugin snapshot to https://repo.jenkins-ci.org/snapshots/fr/inria/repairnator
+    cd repairnator-jenkins-plugin
```
Comment (monperrus) [inline .travis/travis-deploy.sh]: we need to also build the plugin in `.travis/travis-run.sh`

File: repairnator/repairnator-jenkins-plugin/pom.xml [hunk 3]
```diff
@@ -8,13 +8,14 @@
         <relativePath />
     </parent>
     <groupId>fr.inria.repairnator</groupId>
-    <artifactId>repairnator-plugins</artifactId>
-    <version>1.0-SNAPSHOT</version>
+    <artifactId>repairnator</artifactId>
```
Comment (monperrus) [inline repairnator/repairnator-jenkins-plugin/pom.xml]: rename to repairnator-jenkins-plugin

File: repairnator/repairnator-pipeline/src/test/java/fr/inria/spirals/repairnator/process/step/paths/TestComputePlugins.java [hunk 4]
```diff
@@ -76,7 +76,7 @@ public void testComputePluginsWithSingleModuleProject() throws IOException {
             assertThat(stepStatus.isSuccess(), is(true));
         }
 
-        assertTrue(jobStatus.getProperties().getProjectMetrics().getNumberPlugins() >= 7);
```
Comment (monperrus) [inline repairnator/repairnator-pipeline/src/test/java/fr/inria/spirals/repairnator/process/step/paths/TestComputePlugins.java]: here you undo a change we made to fix flakiness, probably a missing merge against master?

File: .travis/travis-deploy.sh (hunk 2 above)
Comment (henry-lp) [reply .travis/travis-deploy.sh]: done :)

File: repairnator/repairnator-pipeline/src/test/java/fr/inria/spirals/repairnator/process/step/paths/TestComputePlugins.java (hunk 4 above)
Comment (henry-lp) [reply repairnator/repairnator-pipeline/src/test/java/fr/inria/spirals/repairnator/process/step/paths/TestComputePlugins.java]: sorry about that, I kind of missed this due to changes in privileges that we now have to commit into fork instead.

File: repairnator/repairnator-jenkins-plugin/pom.xml (hunk 3 above)
Comment (henry-lp) [reply repairnator/repairnator-jenkins-plugin/pom.xml]: done :)

**Comment by henry-lp** [comment]
Ready for review :) . This is a working version for all repairnator tools (except `AssertFixer`) tested on [http://jenkins.kth-assert.net/](http://jenkins.kth-assert.net/)

**Comment by monperrus** [comment]
Thanks a lot. We continue to test it on different Jenkins instances.

**Pull request merged**